rotkehlchen/tasks/manager.py

```diff
@@ -397,25 +406,47 @@ def _maybe_decode_evm_transactions(self) -> None:
     def _maybe_check_premium_status(self) -> None:
         """
         Validates the premium status of the account and if the credentials are not valid
-        it deactivates the user's premium status.
+        it retries 3 times before deactivating the user's premium status. If the
+        credentials are valid and the premium status is not correct it will reactivate
+        the user's premium status.
         """
         now = ts_now()
         if now - self.last_premium_status_check < PREMIUM_STATUS_CHECK:
             return
 
         db_credentials = self.database.get_rotkehlchen_premium()
-        if db_credentials:
-            try:
-                premium_create_and_verify(db_credentials)
-            except PremiumAuthenticationError as e:
-                message = (
-                    f'Could not authenticate with the rotkehlchen server with '
-                    f'the API keys found in the Database. Error: {str(e)}. Will '
-                    f'deactivate the premium status.'
+        if not db_credentials:
+            self.last_premium_status_check = now
+            return
+
+        try:
+            premium_create_and_verify(db_credentials)
+        except PremiumAuthenticationError as e:
```

Comment:
I expect specific handling in network errors here. You handle network error and actual authentication failure the same. 

The idea is that if the server does indeed reply and says that the premium is off, then this is it, it should be off.

Otherwise try 3 more times if it's due to a network error.

Your current approach has a fatal problem. If I no longer have premium, but still have anything in the db credentials then I get free premium for 4 hours. Then I restart the app and get another 4 hours of free premium.